### PR TITLE
Resolve bugs

### LINKED ISSUE #181: [PE-D] The validation of the note added can be improved.
If I want to add a note in this app, I can set both of the title and content to be empty, which in real life should not be a valid note.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: JiawenLyu/ped#2

### LINKED ISSUE #200: [PE-D] Issues with note add (suggestion)
Same as previous issues, you can list the topics that is available as your error message will provide more useful information and user do not need to input topics to see what topics there are.

-------------
Labels: `severity.VeryLow` `type.FeatureFlaw`
original: lingsihui/ped#5

### LINKED ISSUE #201: [PE-D] Issues with note delete error message (suggestion)
Error message shows that it is an invalid topic name, is this an invalid note index ? 

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: lingsihui/ped#6

### LINKED ISSUE #202: [PE-D] Issues with note delete user interface.
Prompts user for a note index when there is no note index available!

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: lingsihui/ped#7

### PR DESCRIPTION
**Pull request by joelngyx**
Resolves #202
Resolves #201 
Resolves #200 
Resolves #181

### CODE REVIEW

File: src/main/java/seedu/eduke8/parser/MenuParser.java [hunk 1]
```diff
@@ -164,14 +164,14 @@ public Command parseCommand(DisplayableList topicList, String userInput) {
             return new BookmarkCommand(BOOKMARK_LIST, bookmarks);
         case COMMAND_NOTE:
             try {
-                if (commandArr[1].equalsIgnoreCase(COMMAND_NOTE_ADD) || commandArr[1]
-                        .equalsIgnoreCase(COMMAND_NOTE_DELETE) || commandArr[1].equals(COMMAND_NOTE_LIST)) {
+                if (commandArr[1].equalsIgnoreCase("add") || commandArr[1]
+                        .equalsIgnoreCase("delete") || commandArr[1].equals("list")) {
```
Comment (josephhhhhhhhh) [inline src/main/java/seedu/eduke8/parser/MenuParser.java]: Would it be possible to use String constants to remove the magic literals?

File: src/main/java/seedu/eduke8/ui/Ui.java [hunk 2]
```diff
@@ -351,33 +381,41 @@ public void addNoteInteractions(TopicList topicList) {
...
+            ui.printError(e.getMessage());
+        }
+
+        if (topicList.doesTopicExist(topicName) && noteCount > 0) {
+            ui.printNoteList(noteList);
+
+            System.out.println(DELETE_NOTE_PROMPT_FOR_INDEX);
+            String input = SCANNER.nextLine();
+
+            if (input.matches("[0-9]+") && Integer.parseInt(input) > 0
```
Comment (josephhhhhhhhh) [inline src/main/java/seedu/eduke8/ui/Ui.java]: Take note the magic literal here, might want to replace it with a String constant!

File: src/main/java/seedu/eduke8/ui/Ui.java [hunk 3]
```diff
@@ -351,33 +381,41 @@ public void addNoteInteractions(TopicList topicList) {
...
+                System.out.println(DELETE_NOTE_SUCCESSFULLY);
             } else {
                 printMessage(INPUT_ERROR + System.lineSeparator() + DELETE_NOTE_UNSUCCESSFULLY);
             }
-        } catch (Eduke8Exception e) {
-            ui.printError(e.getMessage());
+        } else {
+            System.out.println(INVALID_TOPIC);
+            ui.printTopicsError(topicList);
+            System.out.println(INPUT_ERROR + "\n" + DELETE_NOTE_UNSUCCESSFULLY);
```
Comment (josephhhhhhhhh) [inline src/main/java/seedu/eduke8/ui/Ui.java:419]: Would it be possible to use purely the UI Class to output messages?

File: src/main/java/seedu/eduke8/ui/Ui.java [hunk 4]
```diff
@@ -351,33 +381,41 @@ public void addNoteInteractions(TopicList topicList) {
...
+            ui.printNoteList(noteList);
+
+            System.out.println(DELETE_NOTE_PROMPT_FOR_INDEX);
+            String input = SCANNER.nextLine();
+
+            if (input.matches("[0-9]+") && Integer.parseInt(input) > 0
+                    && Integer.parseInt(input) <= noteList.getCount()) {
+                int index = Integer.parseInt(input);
+                topic.getNoteList().delete(index - 1);
+                System.out.println(DELETE_NOTE_SUCCESSFULLY);
```
Comment (josephhhhhhhhh) [inline src/main/java/seedu/eduke8/ui/Ui.java:412]: Would it be possible to use purely the UI Class to output messages?

**Review by josephhhhhhhhh — CHANGES_REQUESTED** [review]
Do take a look at my comments and make the revisions so that this can be merged in!

File: src/main/java/seedu/eduke8/parser/MenuParser.java (hunk 1 above)
Comment (joelngyx) [reply src/main/java/seedu/eduke8/parser/MenuParser.java]: done!

**Review by josephhhhhhhhh — APPROVED** [review]
Overall looks good, will merge it in.

**Pull request merged**